source/application/transforms/rewrite-imports/index.js

```diff
@@ -0,0 +1,40 @@
+import {relative, join, basename} from 'path';
+
+export default function createRewriteImportsTransform (application) {
+	return async function rewriteImportsTransform (file, demo, configuration) {
+		const patternConfig = application.configuration.patterns;
+		const resultName = patternConfig.formats[configuration.outFormat].name;
+
+		const source = file.buffer.toString('utf-8');
+		const resolvePath = configuration.resolve;
+
+		const IDRegistry = Object.keys(file.dependencies).reduce((registry, dependencyName) => {
+			const id = file.dependencies[dependencyName].pattern.id;
+			const path = file.dependencies[dependencyName].pattern.path;
+			return { ...registry, [dependencyName]: { id, path }};
+		}, {});
+
+		function resolveName(localName) {
+			if (IDRegistry[localName]) {
+				const {id, path} = IDRegistry[localName];
+				const treePath = join(...resolvePath(id, resultName, configuration.outFormat));
```

Comment:
It is not clear what resolvePath is, that its a function and why it is taken from configuration instead of an import statement. I think this need to be commented.